Sorting/selectionSort.js

```diff
@@ -0,0 +1,39 @@
+/*  walkthrough to get algorithm straight in my head.
+example unsorted array [1, 3, 100, 0, 12, 2]
+
+1. find the smallest element = 0;
+2. replace the smallest element with the 0th element in the array.
+   [0, 3, 100, 1, 12, 2]
+3. find the smallest element in the unsorted portion of the array = 1
+4. replace that smallest element with the 1st element in the array.
+   [0, 1, 100, 3, 12, 2]
+5. find the smallest element in unsported portion of the array = 2
+   replace that smallest element with the 2nd element in the array.
+   [0, 1, 2, 3, 12, 100]
+*/
+
+function selectionSort(arr) {
+  for (var i = 0; i < arr.length; i++) {
+    //the unsortedPortion starts as the entire array, and decreases in size by 1 every iteration of the loop.
+    var unsortedPortion = arr.slice(i, arr.length)
+    //below is the trickiest line in the program. it's important to add i to the index of the of the smallest element in the unsorted portion
```

Comment:
I believe `slice` creates a new array, so it can confuse the indices, as you point out. Is it possible to avoid creating any new arrays and just use indices into the input array to maintain which portion of the  array is unsorted? I believe that is how it was done in the walkthrough in the comment at the top.